src/main/java/com/gw/web/GeoweaverController.java

```diff
@@ -924,13 +908,27 @@ public ResponseEntity<Resource> fileGetter(ModelMap model, @PathVariable(value="
 			String id = request.getParameter("id");
 			String mode = request.getParameter("mode");
 			String token = request.getParameter("token");
-			String history_id = BaseTool.isNull(request.getParameter("history_id")) ? 
+			String history_id = BaseTool.isNull(request.getParameter("history_id")) ?
 					new RandomString(18).nextString() : request.getParameter("history_id");
 			String[] hosts = request.getParameterValues("hosts[]");
 			String[] encrypted_password = request.getParameterValues("passwords[]");
 			String[] environments = request.getParameterValues("envs[]");
 			String[] passwords = RSAEncryptTool.getPasswords(encrypted_password, session.getId());
 			resp = wt.execute(history_id, id, mode, hosts, passwords, environments, token);
+
+            assert id != null;
+            Optional<Workflow> optionalWorkflow = workflowrepository.findById(id);
+			if (optionalWorkflow.isPresent()) {
+				Workflow workflow = optionalWorkflow.get();
+				Checkpoint checkpoint = new Checkpoint();
```

Comment:
I am not sure this is correct. Spring doesn't allow you to `new` an object and eventually save to the database.

Reply:
we use `new` keyword for entity class to create a new object for that request. 
The new keyword is not used if it is a Bean or if the object is a singleton handled by spring boot in terms of creating the object only once, like for example database connection should be only created once when the application starts and never again. for such things we use dependency injection

in this case, we explicilty want the JPA to create new object to insert it into db and its common. we are doing the same for history and other places in the codebase.
<img width="912" alt="image" src="https://github.com/ESIPFed/Geoweaver/assets/45752727/a30e5889-7392-4b06-a6f7-b7aef85d8963">